File: bench/elixir/bintree.exs
defmodule B do
  def make(0), do: nil
  def make(d), do: {make(d - 1), make(d - 1)}
  def check(nil), do: 1
  def check({l, r}), do: 1 + check(l) + check(r)
end

n = String.to_integer(System.get_env("BENCH_N") || "200")
depth = 12
IO.puts(Enum.reduce(1..n, 0, fn _, acc -> acc + B.check(B.make(depth)) end))

File: bench/erlang/bintree.erl
#!/usr/bin/env escript
%%! -mode(compile)
main(_) ->
    N = bench_n(200), Depth = 12,
    io:format("~w~n", [lists:foldl(fun(_, Acc) -> Acc + check(make(Depth)) end, 0, lists:seq(1, N))]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
make(0) -> nil;
make(D) -> {make(D-1), make(D-1)}.
check(nil) -> 1;
check({L, R}) -> 1 + check(L) + check(R).
